iStore/Login/AuthenticationViewModel.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  LoginViewModel.swift
+//  iStore
+//
+//  Created by Mohamed Sayed on 13.03.24.
+//
+
+import Foundation
+
+final class AuthenticationViewModel: ObservableObject {
+    
+    private let authenticationManager: AuthenticationManager
+    @Published var currentUser: CurrentAuthenticatedUser
+    
+    @Published var showError = false
+    @Published var showProgress = false
+    
+    init(authenticationManager: AuthenticationManager, currentUser: CurrentAuthenticatedUser) {
+        self.authenticationManager = authenticationManager
+        self.currentUser = currentUser
+    }
+    
+    func authenticateUser(userName: String, password: String, completion: @escaping () -> Void) {
+        authenticationManager.authenticate(userName: userName, password: password) { [weak self] result in
+            
+            guard let self = self else {
```

Comment:
You can make it `guard let self` directly. Swift update :D

Reply:
That requires an updated version of xCode.